<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.7.30">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>8&nbsp; Introduction to Probability and Probability Distributions – Quantitative Genetics Graduate Education Module - Spring 2023</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./09-correlation_regression.html" rel="next">
<link href="./07-Rmarkdown.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js" type="module"></script>
<script src="site_libs/quarto-html/tabsets/tabsets.js" type="module"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-de070a7b0ab54f8780927367ac907214.css" rel="stylesheet" class="quarto-color-scheme" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-dark-2b3e328b71be8d25427581baeb23079b.css" rel="stylesheet" class="quarto-color-scheme quarto-color-alternate" id="quarto-text-highlighting-styles">
<link href="site_libs/quarto-html/quarto-syntax-highlighting-de070a7b0ab54f8780927367ac907214.css" rel="stylesheet" class="quarto-color-scheme-extra" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap-249b02670fc29c84004c4c4998805319.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme" id="quarto-bootstrap" data-mode="light">
<link href="site_libs/bootstrap/bootstrap-dark-456d9877428c6b713218490fc9ae1ea2.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme quarto-color-alternate" id="quarto-bootstrap" data-mode="dark">
<link href="site_libs/bootstrap/bootstrap-249b02670fc29c84004c4c4998805319.min.css" rel="stylesheet" append-hash="true" class="quarto-color-scheme-extra" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>

  <script src="https://cdnjs.cloudflare.com/polyfill/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

<link rel="stylesheet" href="style.css">
</head>

<body class="nav-sidebar docked quarto-light"><script id="quarto-html-before-body" type="application/javascript">
    const toggleBodyColorMode = (bsSheetEl) => {
      const mode = bsSheetEl.getAttribute("data-mode");
      const bodyEl = window.document.querySelector("body");
      if (mode === "dark") {
        bodyEl.classList.add("quarto-dark");
        bodyEl.classList.remove("quarto-light");
      } else {
        bodyEl.classList.add("quarto-light");
        bodyEl.classList.remove("quarto-dark");
      }
    }
    const toggleBodyColorPrimary = () => {
      const bsSheetEl = window.document.querySelector("link#quarto-bootstrap:not([rel=disabled-stylesheet])");
      if (bsSheetEl) {
        toggleBodyColorMode(bsSheetEl);
      }
    }
    const setColorSchemeToggle = (alternate) => {
      const toggles = window.document.querySelectorAll('.quarto-color-scheme-toggle');
      for (let i=0; i < toggles.length; i++) {
        const toggle = toggles[i];
        if (toggle) {
          if (alternate) {
            toggle.classList.add("alternate");
          } else {
            toggle.classList.remove("alternate");
          }
        }
      }
    };
    const toggleColorMode = (alternate) => {
      // Switch the stylesheets
      const primaryStylesheets = window.document.querySelectorAll('link.quarto-color-scheme:not(.quarto-color-alternate)');
      const alternateStylesheets = window.document.querySelectorAll('link.quarto-color-scheme.quarto-color-alternate');
      manageTransitions('#quarto-margin-sidebar .nav-link', false);
      if (alternate) {
        // note: dark is layered on light, we don't disable primary!
        enableStylesheet(alternateStylesheets);
        for (const sheetNode of alternateStylesheets) {
          if (sheetNode.id === "quarto-bootstrap") {
            toggleBodyColorMode(sheetNode);
          }
        }
      } else {
        disableStylesheet(alternateStylesheets);
        enableStylesheet(primaryStylesheets)
        toggleBodyColorPrimary();
      }
      manageTransitions('#quarto-margin-sidebar .nav-link', true);
      // Switch the toggles
      setColorSchemeToggle(alternate)
      // Hack to workaround the fact that safari doesn't
      // properly recolor the scrollbar when toggling (#1455)
      if (navigator.userAgent.indexOf('Safari') > 0 && navigator.userAgent.indexOf('Chrome') == -1) {
        manageTransitions("body", false);
        window.scrollTo(0, 1);
        setTimeout(() => {
          window.scrollTo(0, 0);
          manageTransitions("body", true);
        }, 40);
      }
    }
    const disableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        stylesheet.rel = 'disabled-stylesheet';
      }
    }
    const enableStylesheet = (stylesheets) => {
      for (let i=0; i < stylesheets.length; i++) {
        const stylesheet = stylesheets[i];
        if(stylesheet.rel !== 'stylesheet') { // for Chrome, which will still FOUC without this check
          stylesheet.rel = 'stylesheet';
        }
      }
    }
    const manageTransitions = (selector, allowTransitions) => {
      const els = window.document.querySelectorAll(selector);
      for (let i=0; i < els.length; i++) {
        const el = els[i];
        if (allowTransitions) {
          el.classList.remove('notransition');
        } else {
          el.classList.add('notransition');
        }
      }
    }
    const isFileUrl = () => {
      return window.location.protocol === 'file:';
    }
    const hasAlternateSentinel = () => {
      let styleSentinel = getColorSchemeSentinel();
      if (styleSentinel !== null) {
        return styleSentinel === "alternate";
      } else {
        return false;
      }
    }
    const setStyleSentinel = (alternate) => {
      const value = alternate ? "alternate" : "default";
      if (!isFileUrl()) {
        window.localStorage.setItem("quarto-color-scheme", value);
      } else {
        localAlternateSentinel = value;
      }
    }
    const getColorSchemeSentinel = () => {
      if (!isFileUrl()) {
        const storageValue = window.localStorage.getItem("quarto-color-scheme");
        return storageValue != null ? storageValue : localAlternateSentinel;
      } else {
        return localAlternateSentinel;
      }
    }
    const toggleGiscusIfUsed = (isAlternate, darkModeDefault) => {
      const baseTheme = document.querySelector('#giscus-base-theme')?.value ?? 'light';
      const alternateTheme = document.querySelector('#giscus-alt-theme')?.value ?? 'dark';
      let newTheme = '';
      if(authorPrefersDark) {
        newTheme = isAlternate ? baseTheme : alternateTheme;
      } else {
        newTheme = isAlternate ? alternateTheme : baseTheme;
      }
      const changeGiscusTheme = () => {
        // From: https://github.com/giscus/giscus/issues/336
        const sendMessage = (message) => {
          const iframe = document.querySelector('iframe.giscus-frame');
          if (!iframe) return;
          iframe.contentWindow.postMessage({ giscus: message }, 'https://giscus.app');
        }
        sendMessage({
          setConfig: {
            theme: newTheme
          }
        });
      }
      const isGiscussLoaded = window.document.querySelector('iframe.giscus-frame') !== null;
      if (isGiscussLoaded) {
        changeGiscusTheme();
      }
    };
    const authorPrefersDark = false;
    const darkModeDefault = authorPrefersDark;
      document.querySelector('link#quarto-text-highlighting-styles.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
      document.querySelector('link#quarto-bootstrap.quarto-color-scheme-extra').rel = 'disabled-stylesheet';
    let localAlternateSentinel = darkModeDefault ? 'alternate' : 'default';
    // Dark / light mode switch
    window.quartoToggleColorScheme = () => {
      // Read the current dark / light value
      let toAlternate = !hasAlternateSentinel();
      toggleColorMode(toAlternate);
      setStyleSentinel(toAlternate);
      toggleGiscusIfUsed(toAlternate, darkModeDefault);
      window.dispatchEvent(new Event('resize'));
    };
    // Switch to dark mode if need be
    if (hasAlternateSentinel()) {
      toggleColorMode(true);
    } else {
      toggleColorMode(false);
    }
  </script>

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./08-probability_probdists.html"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Introduction to Probability and Probability Distributions</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation docked overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Quantitative Genetics Graduate Education Module - Spring 2023</a> 
        <div class="sidebar-tools-main tools-wide">
    <a href="https://github.com/wcresko/UO_Quant_Gen_Text" title="Source Code" class="quarto-navigation-tool px-1" aria-label="Source Code"><i class="bi bi-github"></i></a>
    <div class="dropdown">
      <a href="" title="Download" id="quarto-navigation-tool-dropdown-0" class="quarto-navigation-tool dropdown-toggle px-1" data-bs-toggle="dropdown" aria-expanded="false" role="link" aria-label="Download"><i class="bi bi-download"></i></a>
      <ul class="dropdown-menu" aria-labelledby="quarto-navigation-tool-dropdown-0">
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="./Quantitative-Genetics-Graduate-Education-Module---Spring-2023.pdf">
              <i class="bi bi-file-pdf pe-1"></i>
            Download PDF
            </a>
          </li>
          <li>
            <a class="dropdown-item sidebar-tools-main-item" href="./Quantitative-Genetics-Graduate-Education-Module---Spring-2023.epub">
              <i class="bi bi-journal pe-1"></i>
            Download ePub
            </a>
          </li>
      </ul>
    </div>
  <a href="" class="quarto-color-scheme-toggle quarto-navigation-tool  px-1" onclick="window.quartoToggleColorScheme(); return false;" title="Toggle dark mode"><i class="bi"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">Course Overview</span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./01-syllabus.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title">Introduction to the course</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./02-schedule.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Course Schedule</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./03-background.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Getting Started with R and Studio</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./04-data_organization.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Organizing and manipulating data files</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./05-R_introduction.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">An Introduction to the R language</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./06-R_morefunc_dfs_plot.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">More R Functions, Complex Objects, Basic Plotting, and RMarkdown</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./07-Rmarkdown.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">A brief Introduction to RMarkdown</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./08-probability_probdists.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Introduction to Probability and Probability Distributions</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./09-correlation_regression.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Correlation and Simple Linear Regression</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./10-ANOVA_intro.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">10</span>&nbsp; <span class="chapter-title">Introduction to Analysis of Variance</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./21-references.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text">References</span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#background" id="toc-background" class="nav-link active" data-scroll-target="#background"><span class="header-section-number">8.1</span> Background</a></li>
  <li><a href="#what-is-probability" id="toc-what-is-probability" class="nav-link" data-scroll-target="#what-is-probability"><span class="header-section-number">8.2</span> What is probability?</a></li>
  <li><a href="#random-variables-probability" id="toc-random-variables-probability" class="nav-link" data-scroll-target="#random-variables-probability"><span class="header-section-number">8.3</span> Random variables &amp; probability</a></li>
  <li><a href="#probability-and-the-bernoulli-distribution" id="toc-probability-and-the-bernoulli-distribution" class="nav-link" data-scroll-target="#probability-and-the-bernoulli-distribution"><span class="header-section-number">8.4</span> Probability and the Bernoulli distribution</a></li>
  <li><a href="#probability-rules" id="toc-probability-rules" class="nav-link" data-scroll-target="#probability-rules"><span class="header-section-number">8.5</span> Probability rules</a></li>
  <li><a href="#joint-probability" id="toc-joint-probability" class="nav-link" data-scroll-target="#joint-probability"><span class="header-section-number">8.6</span> Joint probability</a></li>
  <li><a href="#conditional-probability" id="toc-conditional-probability" class="nav-link" data-scroll-target="#conditional-probability"><span class="header-section-number">8.7</span> Conditional probability</a></li>
  <li><a href="#a-brief-note-on-likelihood-vs.-probability" id="toc-a-brief-note-on-likelihood-vs.-probability" class="nav-link" data-scroll-target="#a-brief-note-on-likelihood-vs.-probability"><span class="header-section-number">8.8</span> A brief note on likelihood vs.&nbsp;probability</a></li>
  <li><a href="#probability-distributions-commonly-used-in-biological-statistics" id="toc-probability-distributions-commonly-used-in-biological-statistics" class="nav-link" data-scroll-target="#probability-distributions-commonly-used-in-biological-statistics"><span class="header-section-number">8.9</span> Probability distributions commonly used in biological statistics</a>
  <ul class="collapse">
  <li><a href="#discrete-probability-distributions" id="toc-discrete-probability-distributions" class="nav-link" data-scroll-target="#discrete-probability-distributions"><span class="header-section-number">8.9.1</span> Discrete Probability Distributions</a></li>
  <li><a href="#continuous-probability-distributions" id="toc-continuous-probability-distributions" class="nav-link" data-scroll-target="#continuous-probability-distributions"><span class="header-section-number">8.9.2</span> <strong>Continuous probability distributions</strong></a></li>
  </ul></li>
  <li><a href="#exercises-associated-with-this-chapter" id="toc-exercises-associated-with-this-chapter" class="nav-link" data-scroll-target="#exercises-associated-with-this-chapter"><span class="header-section-number">8.10</span> Exercises associated with this chapter:</a></li>
  <li><a href="#additional-learning-resources" id="toc-additional-learning-resources" class="nav-link" data-scroll-target="#additional-learning-resources"><span class="header-section-number">8.11</span> Additional learning resources:</a></li>
  </ul>
<div class="toc-actions"><ul><li><a href="https://github.com/wcresko/UO_Quant_Gen_Text/edit/main/08-probability_probdists.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/wcresko/UO_Quant_Gen_Text/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">


<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Introduction to Probability and Probability Distributions</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="background" class="level2" data-number="8.1">
<h2 data-number="8.1" class="anchored" data-anchor-id="background"><span class="header-section-number">8.1</span> Background</h2>
<p>In this chapter, we will cover the basics of probability and common probability distributions. Probabilistic thinking can mark a significant departure in how we typically consider mathematics, and the world around us more generally. We have to shelve our natural inclination toward determinism, and embrace random variables, shades of likelihood, and complexity. As we’ll see, uncertainty in our estimates is a given. Indeed, the process of statistics is largely about quantifying and managing uncertainty - a process that begins with understanding probability distributions.</p>
<p>Frequently, we want to understand how likely a particular observation or set of observations is (e.g.&nbsp;from a sample of a population), given some expectation. That expectation may be based on a theoretical probability distribution we can use to model variation in nature. In this chapter we will introduce some core concepts of probability and how those pertain to understanding observed <strong>parameters</strong>, or features, and variation within systems.</p>
</section>
<section id="what-is-probability" class="level2" data-number="8.2">
<h2 data-number="8.2" class="anchored" data-anchor-id="what-is-probability"><span class="header-section-number">8.2</span> What is probability?</h2>
<p>Statistical probability can be understood from two distinct perspectives: the “Frequentist” and the “Bayesian”.</p>
<ul>
<li><strong>Frequency interpretation</strong> <br> <br> “Probabilities are mathematically convenient approximations to long run relative frequencies.” <br> <br></li>
<li><strong>Subjective (Bayesian) interpretation</strong> <br> <br> “Probability statements are expressions of the opinion of some individual or of current understanding regarding how certain an event is to occur.” <br></li>
</ul>
<p>Both conceptions of probability are widely applied in data analysis, though most of the techniques discussed in this book are rooted in frequentist statistics.</p>
</section>
<section id="random-variables-probability" class="level2" data-number="8.3">
<h2 data-number="8.3" class="anchored" data-anchor-id="random-variables-probability"><span class="header-section-number">8.3</span> Random variables &amp; probability</h2>
<p><br></p>
<p><strong>Probability</strong> is the expression of belief in some future outcome based on information about a system, and is typically applied in statistics to variables we want to understand or estimate in the real world. Specifically, a <strong>random variable</strong> can take on different values at varying probabilities based on its underlying sample space and probability distribution. The <strong>sample space</strong> of a random variable is the universe of all possible values for that variable. It may be helpful to think of the sample space in the form of a plotted function, where possible values of the random variable make up the x-axis, and the probability of “drawing” a particular value at random makes up the y-axis.</p>
<p><br></p>
<p>The <strong>sample space</strong> can be represented by a <strong>probability distribution</strong> when our random variable is discrete. By discrete we mean that the variable can take on a limited (finite) number of values. Meristic traits like the number of bristles on the abdomen of an insect or the number of action potentials a neuron experiences in a single window of time can only have positive integer values. Continuous random variables like human height, on the other hand, can in theory take on an infinite number of values, but are in practice limited by our measurement precision. For continuous variables, the sample space is represented by what we call a <strong>probability density function</strong> (PDF), also called a continuous probability distribution. Probabilities over a sample space <strong>always sum to 1.0</strong>, meaning that all possible values for that random variable are encompassed by its probability distribution, and we use tools from algebra (for probability distributions) and calculus (for probability density functions) to make use of their properties in statistical modeling and inference.</p>
<p><br></p>
<p>Distributions of random variables can be expressed as functions that have <strong>moments</strong>. These moments are metrics of a function’s shape, and these can be estimated. For example the 1st, 2nd, 3rd and 4th moments of a distribution correspond to the mean, variance, skewness, and kurtosis, respectively. For now let’s just consider the first two.</p>
<ul>
<li>The expectation or mean of a random variable X is:</li>
</ul>
<p><span class="math display">\[E[X] = \sum_{\text{all x}}^{}xP(X=x) = \mu\]</span> <br></p>
<ul>
<li>Often we want to know how dispersed the random variable is around its mean</li>
<li>One measure of dispersion is the variance:</li>
</ul>
<p><span class="math display">\[Var(X) = E[X^2] = \sigma^2\]</span> <br></p>
<p>There are many <strong>families</strong> or <strong>forms</strong> of probability distributions, and which ones we apply in statistics depend on the dynamical system we are trying to represent. We will return to the most commonly used ones below. Probability distributions are mathematically defined by features we call <em>parameters</em>, which correspond to the moments pointed out above. The parameters of the functions themselves are used to understand properties of the systems we use the functions to model. For example the normal distribution (also called the Gaussian distribution, depicted by a bell curve), which is probably the most famous distribution in statistics, is characterized by 2 parameters: <span class="math inline">\(mu\)</span> (the mean) and <span class="math inline">\(sigma^{2}\)</span> (the variance). In practical terms, those parameters dictate the central peak or “mode” and the spread (width), respectively.</p>
<p>These parameters are clearly important for us in thinking about the systems we study. For example in biology we often think about random variables as values expressed by individual living things. We may consider, in theory, all possible individuals under a given set of circumstances, and one or more random variables associated with those individuals. In statistics we call this theoretical notion of all individuals a <strong><em>population</em></strong>. If we can assume that a random variable in that population has a particular probability distribution, it opens the door to estimating the aforementioned population parameters from a <strong><em>random sample</em></strong> of that population. Mean height definitely tells us something about the most common values in a population of humans, as does the variability of height among individuals. So you can see how probability distributions, when applied under the appropriate assumptions, help us understand, quantify, and compare random variables in populations. We will further explore how population parameters are estimated from random samples in the next chapter. For now, we will introduce various probability distributions and the random variables they represent.</p>
</section>
<section id="probability-and-the-bernoulli-distribution" class="level2" data-number="8.4">
<h2 data-number="8.4" class="anchored" data-anchor-id="probability-and-the-bernoulli-distribution"><span class="header-section-number">8.4</span> Probability and the Bernoulli distribution</h2>
<p>To think about probability and probability distributions, let’s start with the Bernoulli distribution. It describes the expected outcome of an event with probability <code>p</code>. A simple example of this scenario is the flipping of a coin. If that coin is <strong>fair</strong>, then the probabilities of heads or tails are</p>
<p><br></p>
<p><span class="math display">\[Pr(X=\text{Head}) = \frac{1}{2} = 0.5 = p \]</span></p>
<p><span class="math display">\[Pr(X=\text{Tails}) = \frac{1}{2} = 0.5 = 1 - p \]</span></p>
<p>If the coin isn’t fair then <span class="math inline">\(p \neq 0.5\)</span>. At this point, we don’t know whether our coin is fair or not, so let’s estimate the Bernoulli distribution of our coin flip by flipping our coin 1000 times and visualize the results.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="08-probability_probdists_files/figure-html/unnamed-chunk-1-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="288"></p>
</figure>
</div>
</div>
</div>
<p>We can see that our estimated Bernoulli distribution indicates a coin that is very close to fair (see Chapter 13: Frequency Analysis for how we might test this statistically). Note that the probabilities still sum to 1, a property of all probability distributions. <br> <br> <span class="math display">\[ p + (1-p) = 1 \]</span> <br> <br> The Bernoulli distribution can be used to represent other binary possibilities, like success or failure, “yes” or “no” answers, choosing an allele at a biallelic locus from a population, etc…</p>
</section>
<section id="probability-rules" class="level2" data-number="8.5">
<h2 data-number="8.5" class="anchored" data-anchor-id="probability-rules"><span class="header-section-number">8.5</span> Probability rules</h2>
<p>Let’s take a moment to cover some basic rules of probability regarding the observation of multiple “events”.</p>
<p>Let’s say we flip a fair coin twice. Represent the first flip as ‘X’ and the second flip as ‘Y’. <code>H</code> indicates a Heads and <code>T</code> a Tails. The probability for any given sequence of both flips is</p>
<p><br> <br></p>
<p><span class="math display">\[ Pr(\text{X=H and Y=H}) = p*p = p^2 \]</span> <span class="math display">\[ Pr(\text{X=H and Y=T}) = p*p = p^2 \]</span> <span class="math display">\[ Pr(\text{X=T and Y=H}) = p*p = p^2 \]</span> <span class="math display">\[ Pr(\text{X=T and Y=T}) = p*p = p^2 \]</span></p>
<p><br></p>
<p>While the probability of flipping both an <code>H</code> and <code>T</code> in any order is</p>
<p><br></p>
<p><span class="math display">\[ \text{Pr(X=H and Y=T) or Pr(X=T and Y=H)} = \]</span> <span class="math display">\[ (p*p) + (p*p) = 2p^{2} \]</span> <br> <br></p>
<p>These are the <strong>‘and’</strong> and <strong>‘or’</strong> rules for the probability of multiple events:</p>
<ul>
<li>‘and’ means multiply the probabilities to attain the probability of both events happening</li>
<li>‘or’ means sum the probabilities to attain the probability of either event happening</li>
<li>most probability distributions can be built up from these simple rules</li>
</ul>
</section>
<section id="joint-probability" class="level2" data-number="8.6">
<h2 data-number="8.6" class="anchored" data-anchor-id="joint-probability"><span class="header-section-number">8.6</span> Joint probability</h2>
<p>The joint probability is the probability of two or more outcomes co-occurring. Following the ‘and’ rule,</p>
<p><br> <br> <span class="math display">\[Pr(X,Y) = Pr(X) * Pr(Y)\]</span><br>
<br></p>
<p>As above, this multiplication is true for two <strong>independent</strong> events. However, for two non-independent events we also have to take into account their <strong>covariance</strong>. To do this we need to determine their <strong>conditional probabilities</strong>.</p>
</section>
<section id="conditional-probability" class="level2" data-number="8.7">
<h2 data-number="8.7" class="anchored" data-anchor-id="conditional-probability"><span class="header-section-number">8.7</span> Conditional probability</h2>
<p>Variables that are non-independent have a shared variance, which is also known as <strong>covariance</strong>. You can think of this as two variables that consistently deviate from their respective means. Covariance standardized to a mean of zero and a unit standard deviation is <strong>correlation</strong>, which we’ll discuss in detail in Chapter 12. To assess the probability of two events where they might not be independent, we must considering their conditional probability.</p>
<p><br></p>
<ul>
<li>The conditional probability for two <strong>independent</strong> variables:</li>
</ul>
<p><br></p>
<p><span class="math display">\[Pr(Y|X) = Pr(Y)\]</span> <span class="math display">\[Pr(X|Y) = Pr(X)\]</span> <br></p>
<p>This means that the probability of <code>Y</code> given <code>X</code> is just the probability of <code>Y</code>, and the reverse is true for the probability of <code>X</code> given <code>Y</code>. In other words, the occurrence of event <code>X</code> or <code>Y</code> has no influence on the occurrence of the other event. These variables are therefore independent.</p>
<p><br></p>
<ul>
<li>The conditional probability for two <strong>non-independent</strong> variables:</li>
</ul>
<p><br></p>
<p><span class="math display">\[Pr(Y|X) \neq Pr(Y)\]</span> <span class="math display">\[Pr(X|Y) \neq Pr(X)\]</span> <br></p>
<p>In this case, the probability of <code>Y</code> given <code>X</code> <em>does not</em> equal the probability of just <code>Y</code>. Thus, one is influencing the probability of the other. More specifically, when we have two non-independent events, the equation for the conditional probability of one event given the other is</p>
<p><br></p>
<p><span class="math display">\[Pr(Y|X) = \frac{Pr(X|Y)Pr(Y)}{Pr(X)}\]</span> <br></p>
<p>which is also known as <strong>Bayes’ Theorem</strong>.</p>
</section>
<section id="a-brief-note-on-likelihood-vs.-probability" class="level2" data-number="8.8">
<h2 data-number="8.8" class="anchored" data-anchor-id="a-brief-note-on-likelihood-vs.-probability"><span class="header-section-number">8.8</span> A brief note on likelihood vs.&nbsp;probability</h2>
<ul>
<li><p>The <strong>probability</strong> of an event is the proportion of times that the event would occur if we repeated a random trial over and over again under the same conditions.</p></li>
<li><p>The <strong>likelihood</strong> is the probability of observing a particular set of data or outcome, given a particular parameter value.</p></li>
</ul>
<p><code>L[parameter|data] = Pr[data|parameter]</code></p>
<p>Extending from this, the parameter value at which the likelihood is maximized is called the maximum likelihood estimate (MLE). You don’t need to worry too much about likelihood in this course, but realize that many of our formualae for estimating parameters from data actually produce maximum likelihood estimates. The formula we use to calculate a mean from a sample of observations, for example, produces the maximum likelihood estimate for the population mean from which that sample was taken. The <strong>likelihood function</strong> (for a single parameter) or <strong>likelihood surface</strong> (for multiple parameters) describes the relationship between different parameter values and their likelihood. We can’t always derive convenient equations to obtain maximum likelihood estimates, however, and in those cases we may have to rely on algorithmic searches of “parameter space” to find the MLE.</p>
</section>
<section id="probability-distributions-commonly-used-in-biological-statistics" class="level2" data-number="8.9">
<h2 data-number="8.9" class="anchored" data-anchor-id="probability-distributions-commonly-used-in-biological-statistics"><span class="header-section-number">8.9</span> Probability distributions commonly used in biological statistics</h2>
<p>(Many of these are thanks to Sally Otto at UBC)</p>
<section id="discrete-probability-distributions" class="level3" data-number="8.9.1">
<h3 data-number="8.9.1" class="anchored" data-anchor-id="discrete-probability-distributions"><span class="header-section-number">8.9.1</span> Discrete Probability Distributions</h3>
<section id="geometric-distribution" class="level4" data-number="8.9.1.1">
<h4 data-number="8.9.1.1" class="anchored" data-anchor-id="geometric-distribution"><span class="header-section-number">8.9.1.1</span> <strong>Geometric Distribution</strong></h4>
<p>If a single event has two possible outcomes at probability <code>p</code> and <code>1-p</code>, and is independent of past events (<em>i.e.</em> a Bernoulli trial), the probability of having to observe <code>k</code> trials before the first “success” appears is given by the <strong>geometric distribution</strong>. The probability that the first “success” would appear on the first trial is <code>p</code>, but the probability that the <em>first</em> “success” appears on the second trial is <code>(1-p)*p</code>. By generalizing this procedure, the probability that there will be <code>k-1</code> failures before the first success is:</p>
<p><span class="math display">\[P(X=k)=(1-p)^{k-1}p\]</span></p>
<ul>
<li>mean = <span class="math inline">\(\frac{1}{p}\)</span></li>
<li>variance = <span class="math inline">\(\frac{(1-p)}{p^2}\)</span></li>
</ul>
<section id="the-geometric-distribution-in-practice" class="level5" data-number="8.9.1.1.1">
<h5 data-number="8.9.1.1.1" class="anchored" data-anchor-id="the-geometric-distribution-in-practice"><span class="header-section-number">8.9.1.1.1</span> The <strong>Geometric Distribution</strong> in practice</h5>
<p>The geometric distribution applies in any scenario in which we want to know the probability of a certain number of failures before we observe an event (assuming each trial is independent). Dice rolls, free throws in basketball, sales pitches, and many more such sequential trials with two outcomes are modeled well by the geometric distribution.</p>
<p>For example, if the probability of extinction of an endangered population is estimated to be 0.1 every year, what is the expected time until extinction?</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/prob.017.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:80.0%"></p>
</figure>
</div>
</div>
</div>
<p>Here we can see the probability of extinction in any given year. If we wanted to know the probability of extinction <em>by</em> a specific year, we can simply apply the ‘or’ rule and sum the probabilities up until the specified year. The probability of extinction by year 4 is equivalent to the probability of extinction in year 1 <em>or</em> year 2 <em>or</em> year 3, <span class="math inline">\(0.1 + (1 - 0.1)*0.1 + (1 - 0.1)^2*0.1 = 0.271\)</span>.</p>
</section>
</section>
<section id="binomial-distribution" class="level4" data-number="8.9.1.2">
<h4 data-number="8.9.1.2" class="anchored" data-anchor-id="binomial-distribution"><span class="header-section-number">8.9.1.2</span> <strong>Binomial Distribution</strong></h4>
<p>A <strong>binomial distribution</strong> represents the distribution of outcomes from the <strong>combination</strong> of several Bernoulli trials <em>i.e.</em> independent trials with only two outcomes. In fact, the Bernoulli distribution is just a special case of the binomial distribution for n = 1 Bernoulli trials. The distribution of probabilities for each combination of outcomes is</p>
<p><span class="math display">\[\large f(k) = {n \choose k} p^{k} (1-p)^{n-k}\]</span> - <code>n</code> is the total number of trials - <code>k</code> is the number of successes - <code>p</code> is the probability of success - <code>q</code> is the probability of not success - For binomial as with the Bernoulli <code>p = 1-q</code></p>
<section id="the-binomial-distribution-in-practice" class="level5" data-number="8.9.1.2.1">
<h5 data-number="8.9.1.2.1" class="anchored" data-anchor-id="the-binomial-distribution-in-practice"><span class="header-section-number">8.9.1.2.1</span> The <strong>binomial distribution</strong> in practice</h5>
<p>Pretend that you flip 20 fair coins (or collect alleles from a heterozygote). Now repeat that process 100 times and record the number of heads that show. We expect that most of the time we will get approximately 10 heads in 20 flips. However, sometimes we will get many fewer heads or many more heads. If we plot the frequency of the proportion of “successes”, or heads, we get in each of our 100 replicates, we get the binomial distribution. Because our coin is fair, we can reasonably expect this distribution to center around 0.5.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.003.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
<p>The binomial distribution is the basis for frequency tests when outcomes are binary.</p>
</section>
</section>
<section id="negative-binomial-distribution" class="level4" data-number="8.9.1.3">
<h4 data-number="8.9.1.3" class="anchored" data-anchor-id="negative-binomial-distribution"><span class="header-section-number">8.9.1.3</span> <strong>Negative Binomial Distribution</strong></h4>
<p>The <strong>negative binomial distribution</strong> is an extension of the geometric distribution describing the expected time until not just one success but <code>r</code> “successes” have occurred. Mathematically, it is a generalization of the geometric distribution, where the probability of the <span class="math inline">\(r^{th}\)</span> “success” appearing on the <span class="math inline">\(k^{th}\)</span> trial is:</p>
<p><span class="math display">\[P(X=k)=(\frac{k-1}{r-1})p^{r-1}(1-p)^{k-r}p\]</span></p>
<p><br></p>
<p>which simplifies to</p>
<p><span class="math display">\[P(X=k)=(\frac{k-1}{r-1})p^{r}(1-p)^{k-r}\]</span></p>
<ul>
<li>mean = <span class="math inline">\(\frac{r}{p}\)</span></li>
<li>variance = <span class="math inline">\(r(1-p)/p^2\)</span></li>
</ul>
<p>For example, if a predator must capture 10 prey before it can grow large enough to reproduce, what would be the expected age of onset of reproduction if the probability of capturing a prey on any given day is 0.1?</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/prob.018.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:50.0%"></p>
</figure>
</div>
</div>
</div>
<p>Notice that the variance is quite high (~1000) and the distribution is fairly skewed. Generally, a low probability of success <code>p</code> and a high threshold of successes <code>r</code> leads to a highly dispersed distribution with considerable kurtosis (‘tailedness’).</p>
</section>
<section id="poisson-probability-distribution" class="level4" data-number="8.9.1.4">
<h4 data-number="8.9.1.4" class="anchored" data-anchor-id="poisson-probability-distribution"><span class="header-section-number">8.9.1.4</span> <strong>Poisson Probability Distribution</strong></h4>
<p>Another common situation in biology is when each trial is discrete, but the number of observations of each outcome is observed/counted. Such scenarios are modeled well by the <strong>Poisson distribution</strong>. For example, counts of snails in several plots of land, observations of the firing of a neuron in a unit of time, or count of genes in a genome binned to units of 500 AA. Just like before, you have ‘successes’, but now you count them for each replicate where replicates are now units of area or time. Values can now range from 0 to a large number.</p>
<p>For example, you can examine 1000 genes and count the number of base pairs in the coding region of each gene. What is the probability of observing a gene with ‘r’ bp?</p>
<p><code>Pr(Y=r)</code> is the probability that the number of occurrences of an event <code>y</code> equals a count <code>r</code> in the total number of trials.</p>
<p><br></p>
<p><span class="math display">\[Pr(Y=r) = \frac{e^{-\mu}\mu^r}{r!}\]</span></p>
<p>Note that this is a single parameter function because <span class="math inline">\(\mu = \sigma^2\)</span> - the two together are often just represented by <span class="math inline">\(\lambda\)</span>.</p>
<p><span class="math display">\[Pr(y=r) = \frac{e^{-\lambda}\lambda^r}{r!}\]</span></p>
<p>This means that for a variable that is truly Poisson distributed, the mean and variance should be roughly equal to one another. Variables that are approximately Poisson distributed but have a larger variance than the mean are called ‘overdispersed’, indicating that the observed variance is larger than appropriate for the theoretical distribution. This is quite common in RNA-seq and microbiome data. When overdispersion is a problem in count data, we often use the negative binomial distribution instead because it allows the variance to differ from the mean.</p>
<section id="poisson-probability-distribution-gene-length-by-bins-of-500-nucleotides" class="level5" data-number="8.9.1.4.1">
<h5 data-number="8.9.1.4.1" class="anchored" data-anchor-id="poisson-probability-distribution-gene-length-by-bins-of-500-nucleotides"><span class="header-section-number">8.9.1.4.1</span> Poisson Probability Distribution | gene length by bins of 500 nucleotides</h5>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.004.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:80.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="poisson-probability-distribution-increasing-parameter-values-of-lambda" class="level5" data-number="8.9.1.4.2">
<h5 data-number="8.9.1.4.2" class="anchored" data-anchor-id="poisson-probability-distribution-increasing-parameter-values-of-lambda"><span class="header-section-number">8.9.1.4.2</span> Poisson Probability Distribution | increasing parameter values of <span class="math inline">\(\lambda\)</span></h5>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.005.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:70.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="horse-kick-deaths-in-the-prussian-army" class="level5" data-number="8.9.1.4.3">
<h5 data-number="8.9.1.4.3" class="anchored" data-anchor-id="horse-kick-deaths-in-the-prussian-army"><span class="header-section-number">8.9.1.4.3</span> Horse kick deaths in the Prussian army</h5>
<p>One of the earliest applications of the Poisson distribution was in 1898, when it was used to model the number of soldier deaths from horse kicks in 14 different corps of the Prussian army. As can be seen from the chart below, the Poisson distribution does a remarkable job at modeling these unfortunate events. Indeed, while it is useful for count data in general, it is particularly effective at modeling the distribution of unlikely, independent events.</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="08-probability_probdists_files/figure-html/unnamed-chunk-7-1.png" class="img-fluid quarto-figure quarto-figure-center figure-img" width="672"></p>
</figure>
</div>
</div>
</div>
</section>
</section>
</section>
<section id="continuous-probability-distributions" class="level3" data-number="8.9.2">
<h3 data-number="8.9.2" class="anchored" data-anchor-id="continuous-probability-distributions"><span class="header-section-number">8.9.2</span> <strong>Continuous probability distributions</strong></h3>
<p>Up until this point, we have been looking at <em>discrete</em> probability distributions, where our measurements represent integer or categorical values (event outcomes, counts, etc…) and the probability of a specific observation can be directly quantified. We will now discuss probability density functions (PDFs), better known as <strong>continuous probability distributions</strong>. These represent the distribution of <em>continuous</em> values, from which a random sample can take on an infinite number of values within the range of the distribution (limited by measurement accuracy). As such, unlike discrete probability distributions, the probability of finding any <em>exact</em> value within a continuous distribution is effectively 0. We must instead look at the probability of a measurement falling between a range of values, <code>a</code> and <code>b</code> <em>i.e.</em> the integral of the density function (the area beneath the curve) between said values.</p>
<p><br></p>
<p>P(observation lies within dx of x) = f(x)dx</p>
<p><span class="math display">\[P(a\leq X \leq b) = \int_{a}^{b} f(x) dx\]</span></p>
<p><br></p>
<p>Remember that the indefinite integral sums to one</p>
<p><span class="math display">\[\int_{-\infty}^{\infty} f(x) dx = 1\]</span></p>
<p><br></p>
<p>The expected value of a random variable <code>X</code>, <code>E[X]</code>, may be found by integrating the product of <code>x</code> and the probability density function over all possible values of <code>x</code>:</p>
<p><span class="math display">\[E[X] = \int_{-\infty}^{\infty} xf(x) dx \]</span></p>
<p><br></p>
<p><span class="math inline">\(Var(X) = E[X^2] - (E[X])^2\)</span>, where the expectation of <span class="math inline">\(X^2\)</span> is</p>
<p><span class="math display">\[E[X^2] = \int_{-\infty}^{\infty} x^2f(x) dx \]</span></p>
<section id="uniform-distribution" class="level4" data-number="8.9.2.1">
<h4 data-number="8.9.2.1" class="anchored" data-anchor-id="uniform-distribution"><span class="header-section-number">8.9.2.1</span> <strong>Uniform Distribution</strong></h4>
<p>The uniform distribution is rectangular, meaning that all values have equal probability between the bounds of the distribution <span class="math inline">\([a,b]\)</span>. Its PDF for an expected value of <code>X</code> is given by</p>
<p><br></p>
<p><span class="math display">\[E[X] = \int_{a}^{b} x\frac{1}{b-a} dx = \frac{(a+b)}{2} \]</span></p>
<p><br></p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/prob.019.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
<p>While we are introducing it here as a continuous probability distribution, the uniform distribution has a discrete application as well. Both are used to describe any scenario in which an outcome has equal probability to occur, like true random number generation, or a dice roll in the discrete case.</p>
</section>
<section id="exponential-distribution" class="level4" data-number="8.9.2.2">
<h4 data-number="8.9.2.2" class="anchored" data-anchor-id="exponential-distribution"><span class="header-section-number">8.9.2.2</span> <strong>Exponential Distribution</strong></h4>
<p>The <strong>exponential distribution</strong> can be thought of as the continuous alternative to the geometric distribution, describing the probability of the occurrence of an event or state change over time, given a continuous process. It is defined by a single parameter, the rate constant <span class="math inline">\(\lambda\)</span>, which represents the instantaneous probability of an event occurring. The PDF is</p>
<p><br></p>
<p><span class="math display">\[f(x)=\lambda e^{-\lambda x}\]</span></p>
<p><code>E[X]</code> can be found be integrating <span class="math inline">\(xf(x)\)</span> from 0 to infinity, leading to the result that</p>
<p><br></p>
<ul>
<li><span class="math inline">\(E[X] = \frac{1}{\lambda}\)</span></li>
<li><span class="math inline">\(E[X^2] = \frac{1}{\lambda^2}\)</span></li>
</ul>
<p>For example, let <span class="math inline">\(\lambda\)</span> represent the instantaneous death rate of an individual. The expected lifespan of that individual would be described by an exponential distribution (assuming that <span class="math inline">\(\lambda\)</span> does not change over time).</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/prob.020.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:70.0%"></p>
</figure>
</div>
</div>
</div>
<p>More generally, the exponential distribution describes many situations in which the probability of an event is approximately constant and independent. It is widely applied in survival analysis, actuarial sciences, marketing, and the physical sciences (particularly any process exhibiting exponential decay).</p>
</section>
<section id="gamma-distribution" class="level4" data-number="8.9.2.3">
<h4 data-number="8.9.2.3" class="anchored" data-anchor-id="gamma-distribution"><span class="header-section-number">8.9.2.3</span> <strong>Gamma Distribution</strong></h4>
<p>The gamma distribution generalizes the exponential distribution in the same way that the negative binomial distribution generalizes the geometric distribution. Instead of representing the probability of the first occurrence of an event, it models the waiting time until the <span class="math inline">\(r^{th}\)</span> event for a process that occurs randomly over time at a rate <span class="math inline">\(\lambda\)</span>:</p>
<p><br></p>
<p><span class="math display">\[f(x) = \frac{e^{-\lambda x}\lambda x^{r-1}}{(r-1)!}\lambda\]</span></p>
<p><br></p>
<p><span class="math display">\[ Mean =  \frac{r}{\lambda} \]</span> <span class="math display">\[ Variance = \frac{r}{\lambda^2} \]</span></p>
<p><br></p>
<p>For example, if in a PCR reaction, DNA polymerase synthesizes new DNA strands at a rate of 1 per millisecond, how long until 1000 new DNA strands are produced? Assume that DNA synthesis does not deplete the pool of primers or nucleotides in the chamber, so that each event is independent of other events in the PCR chamber.</p>
</section>
<section id="the-gaussian-or-normal-distribution" class="level4" data-number="8.9.2.4">
<h4 data-number="8.9.2.4" class="anchored" data-anchor-id="the-gaussian-or-normal-distribution"><span class="header-section-number">8.9.2.4</span> The Gaussian or Normal Distribution</h4>
<p>The ‘Gaussian’, or <strong>Normal distribution</strong> is one of the best known probability distributions. Many people whether statistically versed or not have an intuitive understanding of the normal distribution because it models the nature of random continuous variables in a population well - that is, they have a central tendency plus a constrained amount of deviation around this tendency (a “bell curve”). The normal distribution has two parameters, the mean <span class="math inline">\(\mu\)</span> and the standard deviation <span class="math inline">\(\sigma\)</span>. The PDF of the normal distribution is defined as</p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.032.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:40.0%"></p>
</figure>
</div>
</div>
</div>
<p>where <span class="math display">\[\large \pi \approx 3.14159\]</span></p>
<p><span class="math display">\[\large e \approx 2.71828\]</span></p>
<p>To write that a variable (v) is distributed as a normal distribution with mean <span class="math inline">\(\mu\)</span> and variance <span class="math inline">\(\sigma^2\)</span>, we write the following:</p>
<p><span class="math display">\[\large v \sim \mathcal{N} (\mu,\sigma^2)\]</span></p>
<section id="normal-pdf-estimates-of-mean-and-variance" class="level5 smaller" data-number="8.9.2.4.1">
<h5 class="smaller anchored" data-number="8.9.2.4.1" data-anchor-id="normal-pdf-estimates-of-mean-and-variance"><span class="header-section-number">8.9.2.4.1</span> Normal PDF | estimates of mean and variance</h5>
<p>Estimate of the mean from a single sample</p>
<p><span class="math display">\[\Large \bar{x} = \frac{1}{n}\sum_{i=1}^{n}{x_i} \]</span></p>
<p>Estimate of the variance from a single sample</p>
<p><span class="math display">\[\Large s^2 = \frac{1}{n-1}\sum_{i=1}^{n}{(x_i - \bar{x})^2} \]</span></p>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.010.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="why-is-the-normal-distribution-special-in-biology" class="level5" data-number="8.9.2.4.2">
<h5 data-number="8.9.2.4.2" class="anchored" data-anchor-id="why-is-the-normal-distribution-special-in-biology"><span class="header-section-number">8.9.2.4.2</span> Why is the Normal distribution special in biology?</h5>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.013.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.015.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:100.0%"></p>
</figure>
</div>
</div>
</div>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.014.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:60.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="parent-offspring-resemblance" class="level5 smaller" data-number="8.9.2.4.3">
<h5 class="smaller anchored" data-number="8.9.2.4.3" data-anchor-id="parent-offspring-resemblance"><span class="header-section-number">8.9.2.4.3</span> Parent-offspring resemblance</h5>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.016.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:45.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="genetic-model-of-complex-traits" class="level5 smaller" data-number="8.9.2.4.4">
<h5 class="smaller anchored" data-number="8.9.2.4.4" data-anchor-id="genetic-model-of-complex-traits"><span class="header-section-number">8.9.2.4.4</span> Genetic model of complex traits</h5>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.017.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:90.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="distribution-of-f_2-genotypes-really-just-binomial-sampling" class="level5 smaller" data-number="8.9.2.4.5">
<h5 class="smaller anchored" data-number="8.9.2.4.5" data-anchor-id="distribution-of-f_2-genotypes-really-just-binomial-sampling"><span class="header-section-number">8.9.2.4.5</span> Distribution of <span class="math inline">\(F_2\)</span> genotypes | really just binomial sampling</h5>
<div class="cell" data-layout-align="center">
<div class="cell-output-display">
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="images/week_2.018.jpeg" class="img-fluid quarto-figure quarto-figure-center figure-img" style="width:70.0%"></p>
</figure>
</div>
</div>
</div>
</section>
<section id="the-central-limit-theorm" class="level5" data-number="8.9.2.4.6">
<h5 data-number="8.9.2.4.6" class="anchored" data-anchor-id="the-central-limit-theorm"><span class="header-section-number">8.9.2.4.6</span> The Central Limit Theorm</h5>
<p>The normal distribution is immensely useful because of the <strong>Central Limit Theorem</strong>, which says that the mean or the sum of many random variables independently drawn from the same distribution is distributed approximately normally. One can think of numerous real-world situations in which this applies, such as when multiple genes contribute to a phenotype or when many factors contribute to a biological process. In addition, whenever there is variance introduced by stochastic factors the central limit theorem holds. Thus, normal distributions occur throughout genomics, while representing the basis of classical statistics.</p>
</section>
<section id="a-note-on-z-scores-of-normal-variables" class="level5" data-number="8.9.2.4.7">
<h5 data-number="8.9.2.4.7" class="anchored" data-anchor-id="a-note-on-z-scores-of-normal-variables"><span class="header-section-number">8.9.2.4.7</span> A note on z-scores of normal variables</h5>
<p>Often we want to make variables more directly comparable to one another, particularly when they have scales differing by one or more orders of magnitude. For example, consider measuring the leg length of mice and of elephants. Which animal has longer legs in absolute terms? What about proportional to their body size? A good way to answer these last questions is to use ‘z-scores’.</p>
<p>A z-score is a statistic standardized to a mean of 0 and a standard deviation of 1. To attain the z-score for a given estimate, we can modify any normal distribution to have a mean of 0 and a standard deviation of 1 by normalizing the distribution to the population’s standard deviation (another term for this is the standard normal distribution). The z-score, then, is the number of standard deviations from the mean (0) of this distribution.</p>
<p><br></p>
<p><span class="math display">\[\huge z_i = \frac{(x_i - \bar{x})}{s}\]</span></p>
</section>
</section>
</section>
</section>
<section id="exercises-associated-with-this-chapter" class="level2" data-number="8.10">
<h2 data-number="8.10" class="anchored" data-anchor-id="exercises-associated-with-this-chapter"><span class="header-section-number">8.10</span> Exercises associated with this chapter:</h2>
<ul>
<li>Problem Set 2</li>
</ul>
</section>
<section id="additional-learning-resources" class="level2" data-number="8.11">
<h2 data-number="8.11" class="anchored" data-anchor-id="additional-learning-resources"><span class="header-section-number">8.11</span> Additional learning resources:</h2>
<ul>
<li><p>Irizarry, R. A. Introduction to Data Science. https://rafalab.github.io/dsbook/ - A gitbook written by a statistician, with great introductions to key topics in statistical inference.</p></li>
<li><p>Logan, M. 2010. Biostatistical Design and Analysis Using R. - A great intro to R for statistical analysis</p></li>
</ul>


</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
  window.document.addEventListener("DOMContentLoaded", function (event) {
    // Ensure there is a toggle, if there isn't float one in the top right
    if (window.document.querySelector('.quarto-color-scheme-toggle') === null) {
      const a = window.document.createElement('a');
      a.classList.add('top-right');
      a.classList.add('quarto-color-scheme-toggle');
      a.href = "";
      a.onclick = function() { try { window.quartoToggleColorScheme(); } catch {} return false; };
      const i = window.document.createElement("i");
      i.classList.add('bi');
      a.appendChild(i);
      window.document.body.appendChild(a);
    }
    setColorSchemeToggle(hasAlternateSentinel())
    const icon = "";
    const anchorJS = new window.AnchorJS();
    anchorJS.options = {
      placement: 'right',
      icon: icon
    };
    anchorJS.add('.anchored');
    const isCodeAnnotation = (el) => {
      for (const clz of el.classList) {
        if (clz.startsWith('code-annotation-')) {                     
          return true;
        }
      }
      return false;
    }
    const onCopySuccess = function(e) {
      // button target
      const button = e.trigger;
      // don't keep focus
      button.blur();
      // flash "checked"
      button.classList.add('code-copy-button-checked');
      var currentTitle = button.getAttribute("title");
      button.setAttribute("title", "Copied!");
      let tooltip;
      if (window.bootstrap) {
        button.setAttribute("data-bs-toggle", "tooltip");
        button.setAttribute("data-bs-placement", "left");
        button.setAttribute("data-bs-title", "Copied!");
        tooltip = new bootstrap.Tooltip(button, 
          { trigger: "manual", 
            customClass: "code-copy-button-tooltip",
            offset: [0, -8]});
        tooltip.show();    
      }
      setTimeout(function() {
        if (tooltip) {
          tooltip.hide();
          button.removeAttribute("data-bs-title");
          button.removeAttribute("data-bs-toggle");
          button.removeAttribute("data-bs-placement");
        }
        button.setAttribute("title", currentTitle);
        button.classList.remove('code-copy-button-checked');
      }, 1000);
      // clear code selection
      e.clearSelection();
    }
    const getTextToCopy = function(trigger) {
        const codeEl = trigger.previousElementSibling.cloneNode(true);
        for (const childEl of codeEl.children) {
          if (isCodeAnnotation(childEl)) {
            childEl.remove();
          }
        }
        return codeEl.innerText;
    }
    const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
      text: getTextToCopy
    });
    clipboard.on('success', onCopySuccess);
    if (window.document.getElementById('quarto-embedded-source-code-modal')) {
      const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
        text: getTextToCopy,
        container: window.document.getElementById('quarto-embedded-source-code-modal')
      });
      clipboardModal.on('success', onCopySuccess);
    }
      var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
      var mailtoRegex = new RegExp(/^mailto:/);
        var filterRegex = new RegExp('/' + window.location.host + '/');
      var isInternal = (href) => {
          return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
      }
      // Inspect non-navigation links and adorn them if external
     var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
      for (var i=0; i<links.length; i++) {
        const link = links[i];
        if (!isInternal(link.href)) {
          // undo the damage that might have been done by quarto-nav.js in the case of
          // links that we want to consider external
          if (link.dataset.originalHref !== undefined) {
            link.href = link.dataset.originalHref;
          }
        }
      }
    function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
      const config = {
        allowHTML: true,
        maxWidth: 500,
        delay: 100,
        arrow: false,
        appendTo: function(el) {
            return el.parentElement;
        },
        interactive: true,
        interactiveBorder: 10,
        theme: 'quarto',
        placement: 'bottom-start',
      };
      if (contentFn) {
        config.content = contentFn;
      }
      if (onTriggerFn) {
        config.onTrigger = onTriggerFn;
      }
      if (onUntriggerFn) {
        config.onUntrigger = onUntriggerFn;
      }
      window.tippy(el, config); 
    }
    const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
    for (var i=0; i<noterefs.length; i++) {
      const ref = noterefs[i];
      tippyHover(ref, function() {
        // use id or data attribute instead here
        let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
        try { href = new URL(href).hash; } catch {}
        const id = href.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note) {
          return note.innerHTML;
        } else {
          return "";
        }
      });
    }
    const xrefs = window.document.querySelectorAll('a.quarto-xref');
    const processXRef = (id, note) => {
      // Strip column container classes
      const stripColumnClz = (el) => {
        el.classList.remove("page-full", "page-columns");
        if (el.children) {
          for (const child of el.children) {
            stripColumnClz(child);
          }
        }
      }
      stripColumnClz(note)
      if (id === null || id.startsWith('sec-')) {
        // Special case sections, only their first couple elements
        const container = document.createElement("div");
        if (note.children && note.children.length > 2) {
          container.appendChild(note.children[0].cloneNode(true));
          for (let i = 1; i < note.children.length; i++) {
            const child = note.children[i];
            if (child.tagName === "P" && child.innerText === "") {
              continue;
            } else {
              container.appendChild(child.cloneNode(true));
              break;
            }
          }
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(container);
          }
          return container.innerHTML
        } else {
          if (window.Quarto?.typesetMath) {
            window.Quarto.typesetMath(note);
          }
          return note.innerHTML;
        }
      } else {
        // Remove any anchor links if they are present
        const anchorLink = note.querySelector('a.anchorjs-link');
        if (anchorLink) {
          anchorLink.remove();
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        if (note.classList.contains("callout")) {
          return note.outerHTML;
        } else {
          return note.innerHTML;
        }
      }
    }
    for (var i=0; i<xrefs.length; i++) {
      const xref = xrefs[i];
      tippyHover(xref, undefined, function(instance) {
        instance.disable();
        let url = xref.getAttribute('href');
        let hash = undefined; 
        if (url.startsWith('#')) {
          hash = url;
        } else {
          try { hash = new URL(url).hash; } catch {}
        }
        if (hash) {
          const id = hash.replace(/^#\/?/, "");
          const note = window.document.getElementById(id);
          if (note !== null) {
            try {
              const html = processXRef(id, note.cloneNode(true));
              instance.setContent(html);
            } finally {
              instance.enable();
              instance.show();
            }
          } else {
            // See if we can fetch this
            fetch(url.split('#')[0])
            .then(res => res.text())
            .then(html => {
              const parser = new DOMParser();
              const htmlDoc = parser.parseFromString(html, "text/html");
              const note = htmlDoc.getElementById(id);
              if (note !== null) {
                const html = processXRef(id, note);
                instance.setContent(html);
              } 
            }).finally(() => {
              instance.enable();
              instance.show();
            });
          }
        } else {
          // See if we can fetch a full url (with no hash to target)
          // This is a special case and we should probably do some content thinning / targeting
          fetch(url)
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.querySelector('main.content');
            if (note !== null) {
              // This should only happen for chapter cross references
              // (since there is no id in the URL)
              // remove the first header
              if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
                note.children[0].remove();
              }
              const html = processXRef(null, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      }, function(instance) {
      });
    }
        let selectedAnnoteEl;
        const selectorForAnnotation = ( cell, annotation) => {
          let cellAttr = 'data-code-cell="' + cell + '"';
          let lineAttr = 'data-code-annotation="' +  annotation + '"';
          const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
          return selector;
        }
        const selectCodeLines = (annoteEl) => {
          const doc = window.document;
          const targetCell = annoteEl.getAttribute("data-target-cell");
          const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
          const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
          const lines = annoteSpan.getAttribute("data-code-lines").split(",");
          const lineIds = lines.map((line) => {
            return targetCell + "-" + line;
          })
          let top = null;
          let height = null;
          let parent = null;
          if (lineIds.length > 0) {
              //compute the position of the single el (top and bottom and make a div)
              const el = window.document.getElementById(lineIds[0]);
              top = el.offsetTop;
              height = el.offsetHeight;
              parent = el.parentElement.parentElement;
            if (lineIds.length > 1) {
              const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
              const bottom = lastEl.offsetTop + lastEl.offsetHeight;
              height = bottom - top;
            }
            if (top !== null && height !== null && parent !== null) {
              // cook up a div (if necessary) and position it 
              let div = window.document.getElementById("code-annotation-line-highlight");
              if (div === null) {
                div = window.document.createElement("div");
                div.setAttribute("id", "code-annotation-line-highlight");
                div.style.position = 'absolute';
                parent.appendChild(div);
              }
              div.style.top = top - 2 + "px";
              div.style.height = height + 4 + "px";
              div.style.left = 0;
              let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
              if (gutterDiv === null) {
                gutterDiv = window.document.createElement("div");
                gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
                gutterDiv.style.position = 'absolute';
                const codeCell = window.document.getElementById(targetCell);
                const gutter = codeCell.querySelector('.code-annotation-gutter');
                gutter.appendChild(gutterDiv);
              }
              gutterDiv.style.top = top - 2 + "px";
              gutterDiv.style.height = height + 4 + "px";
            }
            selectedAnnoteEl = annoteEl;
          }
        };
        const unselectCodeLines = () => {
          const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
          elementsIds.forEach((elId) => {
            const div = window.document.getElementById(elId);
            if (div) {
              div.remove();
            }
          });
          selectedAnnoteEl = undefined;
        };
          // Handle positioning of the toggle
      window.addEventListener(
        "resize",
        throttle(() => {
          elRect = undefined;
          if (selectedAnnoteEl) {
            selectCodeLines(selectedAnnoteEl);
          }
        }, 10)
      );
      function throttle(fn, ms) {
      let throttle = false;
      let timer;
        return (...args) => {
          if(!throttle) { // first call gets through
              fn.apply(this, args);
              throttle = true;
          } else { // all the others get throttled
              if(timer) clearTimeout(timer); // cancel #2
              timer = setTimeout(() => {
                fn.apply(this, args);
                timer = throttle = false;
              }, ms);
          }
        };
      }
        // Attach click handler to the DT
        const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
        for (const annoteDlNode of annoteDls) {
          annoteDlNode.addEventListener('click', (event) => {
            const clickedEl = event.target;
            if (clickedEl !== selectedAnnoteEl) {
              unselectCodeLines();
              const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
              if (activeEl) {
                activeEl.classList.remove('code-annotation-active');
              }
              selectCodeLines(clickedEl);
              clickedEl.classList.add('code-annotation-active');
            } else {
              // Unselect the line
              unselectCodeLines();
              clickedEl.classList.remove('code-annotation-active');
            }
          });
        }
    const findCites = (el) => {
      const parentEl = el.parentElement;
      if (parentEl) {
        const cites = parentEl.dataset.cites;
        if (cites) {
          return {
            el,
            cites: cites.split(' ')
          };
        } else {
          return findCites(el.parentElement)
        }
      } else {
        return undefined;
      }
    };
    var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
    for (var i=0; i<bibliorefs.length; i++) {
      const ref = bibliorefs[i];
      const citeInfo = findCites(ref);
      if (citeInfo) {
        tippyHover(citeInfo.el, function() {
          var popup = window.document.createElement('div');
          citeInfo.cites.forEach(function(cite) {
            var citeDiv = window.document.createElement('div');
            citeDiv.classList.add('hanging-indent');
            citeDiv.classList.add('csl-entry');
            var biblioDiv = window.document.getElementById('ref-' + cite);
            if (biblioDiv) {
              citeDiv.innerHTML = biblioDiv.innerHTML;
            }
            popup.appendChild(citeDiv);
          });
          return popup.innerHTML;
        });
      }
    }
  });
  </script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./07-Rmarkdown.html" class="pagination-link" aria-label="A brief Introduction to RMarkdown">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">A brief Introduction to RMarkdown</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./09-correlation_regression.html" class="pagination-link" aria-label="Correlation and Simple Linear Regression">
        <span class="nav-page-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Correlation and Simple Linear Regression</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
<p>Quantitative Genetics GEM - University of Oregon</p>
</div>   
    <div class="nav-footer-center">
      &nbsp;
    <div class="toc-actions d-sm-block d-md-none"><ul><li><a href="https://github.com/wcresko/UO_Quant_Gen_Text/edit/main/08-probability_probdists.qmd" class="toc-action"><i class="bi bi-github"></i>Edit this page</a></li><li><a href="https://github.com/wcresko/UO_Quant_Gen_Text/issues/new" class="toc-action"><i class="bi empty"></i>Report an issue</a></li></ul></div></div>
    <div class="nav-footer-right">
<p>Built with <a href="https://quarto.org/">Quarto</a></p>
</div>
  </div>
</footer>




</body></html>